backend/contracts/OasisReward.sol

```diff
@@ -1,19 +1,31 @@
 // SPDX-License-Identifier: MIT
 pragma solidity ^0.8.0;
 
-import {ERC721, ERC721Enumerable} from "@openzeppelin/contracts/token/ERC721/extensions/ERC721Enumerable.sol";
-import {Base64} from "@openzeppelin/contracts/utils/Base64.sol";
-
+import {
+    ERC721, ERC721Enumerable
+} from "@openzeppelin/contracts/token/ERC721/extensions/ERC721Enumerable.sol";
+import {
+    Base64
+} from "@openzeppelin/contracts/utils/Base64.sol";
+
+/**
+ * @title OasisReward
+ * @notice An ERC721 contract that allows minting of NFT rewards for 
+    Oasis Quizzes.
+ * Contains functions to mint new NFTs, generate SVG images, 
+    and retrieve token data.
+ */
 contract OasisReward is ERC721Enumerable {
-    // Mapping from token ID to IPFS URI
-    mapping(uint256 => string) private _tokenURIs;
+    // Mapping from token ID to json metadata index
+    mapping(uint256 => uint256) private _hashIDs;
 
     // Mapping from user address to list of owned token IDs
     mapping(address => uint256[]) private _ownedTokens;
     // Whitelisted msg.sender accounts
     mapping(address => bool) private _allowMint;
+    // Mapping from json metadata index to json metadata string
+    string[] private _tokenURIs;
```

Comment:
This should not be a list, but a mapping from the Token's metadata JSON hash => actual JSON right?

```suggestion
    /// Mapping from JSON hash to JSON.
    mapping(bytes32 => string) private _tokenURIs;
```

Reply:
Changed to mapping.